### PR TITLE
Volunteer signup emails

### PR DESCRIPTION
**Pull request by jkeat**
Links
-----
- Dependency PRs: https://github.com/UPchieve/server/pull/376

Description
-----------
- worker for sending ready-to-coach email
- emails for
    - photo id rejected
    - approved-but-not-onboarded
    - ready to coach
- separate welcome emails for partner vs regular volunteers

todo:
- volunteers dont show up on review page unless background info completed

Developer self-review checklist
-------------------------------
- [ ] Potentially confusing code has been explained with comments
- [ ] No warnings or errors have been introduced; all known error cases have been handled
- [ ] Any appropriate documentation (within the code, README.md, docs, etc) has been updated
- [ ] All edge cases have been addressed

### CODE REVIEW

File: services/MailService.js [hunk 1]
```diff
@@ -31,6 +31,11 @@ const sendEmail = (
     asm
   }
 
+  if (process.env.NODE_ENV === 'test') {
```
Comment (treykris) [inline services/MailService.js:34]: We can add `jest.mock('../../../services/MailService');` to the top of our `UserService.test.ts` file to mock the `MailService` so that the tests don't run the actual MailService functions (similar to our `register.test.ts`)

Comment (jkeat) [reply services/MailService.js:34]: good call!

**Comment by jkeat** [comment]
@treykris i merged together our `updatePendingVolunteerStatus` changes, let me know if the conflict resolution looks good

File: worker/jobs/emailReadyToCoach.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import VolunteerModel from '../../models/Volunteer';
+import { Volunteer } from '../../models/types';
+import MailService from '../../services/SessionService';
```
Comment (treykris) [inline worker/jobs/emailReadyToCoach.ts:3]: Import typo to SessionService

File: worker/jobs/emailReadyToCoach.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  const volunteers = (await VolunteerModel.find({
+    isOnboarded: true,
+    isApproved: true,
+    sentReadyToCoachEmail: false
+  })
+    .lean()
+    .exec()) as Volunteer[];
+
+  for (const volunteer of volunteers) {
+    await MailService.sendReadyToCoachEmail(volunteer);
```
Comment (treykris) [inline worker/jobs/emailReadyToCoach.ts:18]: We should set `sentReadyToCoachEmail` to true here too

File: services/UserService.js [hunk 4]
```diff
@@ -193,8 +194,21 @@ module.exports = {
...
+      volunteerBeforeUpdate.photoIdStatus !== PHOTO_ID_STATUS.REJECTED
+    ) {
       UserActionCtrl.rejectedPhotoId(volunteerId)
+      await MailService.sendPhotoRejectedEmail(volunteerBeforeUpdate)
+    }
+
+    const isNewlyApproved = isApproved && !volunteerBeforeUpdate.isApproved
+    if (isNewlyApproved && !volunteerBeforeUpdate.isOnboarded) {
+      UserActionCtrl.accountApproved(volunteerId)
+      await MailService.sendApprovedNotOnboardedEmail(volunteerBeforeUpdate)
```
Comment (treykris) [inline services/UserService.js:210]: We'll also need to fire the `UserActionCtrl.accountApproved(volunteerId)`  off when the user is onboarded and newly approved

We'll probably need to send off this mail in `addBackgroundInfo` too since right now an open volunteer can have their photo and references approved but still have yet to submit their background info

Comment (jkeat) [reply services/UserService.js:210]: Ahh yeah good point 👍

**Comment by jkeat** [comment]
@treykris Going to use this PR to address this: https://upchieve.monday.com/boards/438837796/pulses/651953228
> volunteers dont show up on review page unless background info completed

which means there will only be one place for approval to happen, the admin route
